fe2-android/app/src/main/java/com/github/dedis/popstellar/utility/ActivityUtils.java

```diff
@@ -23,4 +39,69 @@ public static void replaceFragmentInActivity(
     transaction.replace(frameId, fragment);
     transaction.commit();
   }
+
+  public static boolean storePersistentData(Context context, PersistentData data) {
+    Log.d(TAG, "Initiating storage of " + data);
+
+    try (ObjectOutputStream oos =
+        new ObjectOutputStream(
+            context.openFileOutput(PERSISTENT_DATA_FILE_NAME, Context.MODE_PRIVATE))) {
+      oos.writeObject(data);
+    } catch (IOException e) {
+      ErrorUtils.logAndShow(context, TAG, e, R.string.error_storing_data);
+      return false;
+    }
+    Log.d(TAG, "storage successful");
+    return true;
+  }
+
+  public static PersistentData loadPersistentData(Context context) {
+    Log.d(TAG, "Initiating loading of data");
+
+    PersistentData persistentData;
+    try (ObjectInputStream ois =
+        new ObjectInputStream(context.openFileInput(PERSISTENT_DATA_FILE_NAME))) {
+      persistentData = (PersistentData) ois.readObject();
+    } catch (FileNotFoundException e) {
+      ErrorUtils.logAndShow(context, TAG, e, R.string.nothing_stored);
+      return null;
+    } catch (IOException | ClassNotFoundException e) {
+      ErrorUtils.logAndShow(context, TAG, e, R.string.error_loading_data);
+      return null;
+    }
+
+    Log.d(TAG, "loading of " + persistentData);
+    return persistentData;
+  }
+
+  public static boolean clearStorage(Context context) {
+    Log.d(TAG, "clearing data");
+
+    File file = new File(context.getFilesDir(), PERSISTENT_DATA_FILE_NAME);
+    return file.delete();
+  }
+
+  public static boolean activitySavingRoutine(
+      GlobalNetworkManager networkManager, Wallet wallet, Context context)
+      throws GeneralSecurityException {
+    String serverAddress = networkManager.getCurrentUrl();
+    if (serverAddress == null) {
+      return false;
+    }
+    Set<Channel> subscriptions = networkManager.getMessageSender().getSubscriptions();
+    if (subscriptions == null) {
+      subscriptions = new HashSet<>();
+    }
+
+    String[] seed = wallet.exportSeed();
+    Log.d(
+        TAG,
+        "seed "
+            + Arrays.toString(seed)
```

Comment:
I think logging the seed is a big security issue

Reply:
True, at the time I thought it was not big deal because existing wallet code recorded it. But since then I removed those logs and it is indeed a significant vulnerability. Thanks for catching it !